vespa_analysis/vespa_eval.py

```diff
@@ -32,128 +32,186 @@ def run_mmseq(gen_fasta,ref_fasta,gen_name,threads):
+    return [ref_fasta,ref_name,gen_fasta,gen_name,cache_path,threads]
+
+def optmize_threads(threads):
+    '''
+    This function optmizes the number of threads,
+    since the *Parallel* package requires that the number of jobs/sequences
+    to run in parallel should be divisble by the number of the specified threads
+    '''
+    remainder_thread = 0
+    if len(gen_dict)%threads != 0:
+        for i in range(len(gen_dict)):
+            if (len(gen_dict)-i)%threads == 0:
+                remainder_thread = i
+                break
+    return remainder_thread
+
+def plot(gen_name):
+    df_sav = pd.read_csv('vespa_msa/'+gen_name+'_sav_scores.csv',header=None)
+    df_sav_avg = df_sav.mean()
+    sav_avg_arr = np.array(list(df_sav_avg.values))
+    df_cons = pd.read_csv('vespa_msa/'+gen_name+'_cons_scores.csv',header=None)
+    df_cons_avg = df_cons.mean()
+    cons_avg_arr = np.array(list(df_cons_avg.values))
+    spearman_corr = scipy.stats.mstats.spearmanr(cons_avg_arr,sav_avg_arr)
+    df_all_avg = pd.DataFrame()
+    df_all_avg['Average Mutant SAV Scores'] = list(df_sav_avg.values)
+    df_all_avg['Average Wildtype Conservation Class'] = list(df_cons_avg.values)
+
+    ## Plotting Positional SAV Effect and Conservation Classes ##
+    plt.figure(figsize=(15,8))
+    sns.set(style="darkgrid")
+    sns.lineplot(data=df_all_avg)
+    plt.xlabel('MSA Position Index', fontsize=16)
+    plt.title('Positional SAV Effect and Conservation')
+    plt.legend() 
+    plt.savefig(gen_name+'_sav_cons_score.png')
+
+    ## Plotting Relation Between SAV Effect Scores and Conservation Class Per Position ##
+    plt.figure(figsize=(15,8))
+    sns.set(style="whitegrid")
+    p = sns.regplot(x= list(df_all_avg['Average Mutant SAV Scores'].values),y=list(df_all_avg['Average Wildtype Conservation Class'].values),color="lightseagreen")
+    #calculate slope and intercept of regression equation
+    slope, intercept, r, p, sterr = scipy.stats.linregress(x=p.get_lines()[0].get_xdata(),y=p.get_lines()[0].get_ydata())
+    #add regression equation to plot
+    plt.text(0.25, 7, 'y = ' + str(round(intercept,3)) + ' + (' + str(round(slope,3)) + ')x', fontsize=15, color='teal',fontstyle = "oblique")
+    ## Spearman Correlation Between  Postional Average SAV scores and Average Conservation Classes ##
+    plt.text(0.25, 6, 'Spearman correlation = '+str(round(spearman_corr.statistic,3)), fontsize=15, color='teal',fontstyle = "oblique")
+    plt.xlabel('Positional SAV Scores', fontsize=14)
+    plt.ylabel('Positional Conservation Scores', fontsize=14)
+    plt.savefig(gen_name+'_corr_score.png')
 
-## Parsing Input ##
-ref_fasta = args.r
-ref_name = ref_fasta.split('/')[-1]
-ref_name = ref_name.split('.')[0]
-gen_fasta = args.g 
-gen_name = gen_fasta.split('/')[-1]
-gen_name = gen_name.split('.')[0]
-cache_path = args.cache
-threads = int(args.threads)
 cwd = os.getcwd()
-
+## Parsing Input ##
+args = arg_parser()
+parsed_inputs = parse_input(args)
```

Comment:
You can just directly do this: `ref_fasta, ref_name, gen_fasta, gen_name, cache_path, threads = parse_input(args)`